Lib/test/test_queue.py

```diff
@@ -1009,6 +1009,59 @@ class C:
             gc_collect()  # For PyPy or other GCs.
             self.assertIsNone(wr())
 
+    def test_sizeof(self):
+        # Test that __sizeof__() accounts for underlying data structure
+        q = self.q
+
+        # Get the size of an empty queue
+        empty_size = q.__sizeof__()
+        self.assertGreater(empty_size, 0, "Empty queue should have non-zero size")
+
+        # Size should include basic object structure
+        # For C implementation, this includes the ring buffer array
+        # For Python implementation, this includes the underlying deque
+
+        # Add items within initial capacity (if applicable)
+        for i in range(8):
+            q.put(object())
+
+        size_after_8 = q.__sizeof__()
+        # Size may or may not change depending on implementation
+        # C implementation: no change (still within initial ring buffer capacity)
+        # Python implementation: may change (deque growth, but __sizeof__ may not reflect it)
+
+        # Add one more item to potentially trigger growth
+        q.put(object())  # Now 9 items
+
+        size_after_9 = q.__sizeof__()
+        self.assertGreaterEqual(size_after_9, size_after_8,
+                               "Size should not decrease when adding items")
+
+        # Test with a larger number of items
+        large_q = self.type2test()
+        for i in range(1000):
+            large_q.put(object())
+
+        large_size = large_q.__sizeof__()
+
+        # For C implementation, size should grow with capacity
+        # For Python implementation, __sizeof__ will not account for underlying deque
+        # (this is documented behavior on CPython; PyPy doesn't support __sizeof__ at all)
+        if self.__class__.__name__ == 'CSimpleQueueTest':
```

Comment:
Do not do that. Instead, implement this part of the test in the dedicated teat classes. For the Python implementation, do so in the class below.